pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -180,6 +192,22 @@ public void runValidation() {
     LOGGER.info("Validation completed");
   }
 
+  // If we add a new broker, we want to rebuild the broker resource.
+  void rebuildBrokerResourceWhenBrokerAdded(AbstractTableConfig tableConfig, IdealState brokerIdealState) {
```

Comment:
I was referring to the need for rebuildBrokerResourceWhenBrokerAdded, why cant we simply call    `_pinotHelixResourceManager.rebuildBrokerResourceFromHelixTags(tableName);`. This method probably has the same check that instances in idealstate == instances from tenant config. If not, this logic must go into that method.